dali_tf_plugin/setup.py.in

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from setuptools import setup, find_packages
+from setuptools.command.install import install
+import subprocess
+import os
+import re
+import sys
+import platform
+from shutil import copyfile
+
+import os, fnmatch
+def find(pattern, path):
+    result = []
+    for root, dirs, files in os.walk(path):
+        for name in files:
+            if fnmatch.fnmatch(name, pattern):
+                result.append(os.path.join(root, name))
+    return result
+
+def get_module_path(module_name):
+    module_path = ''
+    for d in sys.path:
+        possible_path = os.path.join(d, module_name)
+        # skip current dir as this is plugin dir
+        if os.path.isdir(possible_path) and len(d) != 0:
+            module_path = possible_path
+            break
+    return module_path
+
+def get_tf_build_flags():
+    tf_cflags = ''
+    tf_lflags = ''
+    try:
+        import tensorflow as tensorflow
+        tf_cflags=" ".join(tensorflow.sysconfig.get_compile_flags())
+        tf_lflags=" ".join(tensorflow.sysconfig.get_link_flags())
+    except:
+        tensorflow_path = get_module_path('tensorflow')
+        if tensorflow_path is not '':
+            tf_cflags=" ".join(["-I" + tensorflow_path + "/include",  "-I" + tensorflow_path + "/include/external/nsync/public", "-D_GLIBCXX_USE_CXX11_ABI=0"])
+            tf_lflags=" ".join(["-L" + tensorflow_path, "-ltensorflow_framework"])
+
+    if tf_cflags is '' and tf_lflags is '':
+        raise ImportError('Could not find Tensorflow. Tensorflow must be installed before installing nvidia-dali-tf-plugin')
+    return (tf_cflags, tf_lflags)
+
+def get_tf_compiler_version():
+    tensorflow_path = get_module_path('tensorflow')
+    tensorflow_libs = find('libtensorflow_framework*so*', tensorflow_path)
+    if not tensorflow_libs:
+        return ''
+    lib = tensorflow_libs[0]
+    cmd = 'strings -a ' + lib + ' | grep "GCC: ("'
+    s = str(subprocess.check_output(cmd, shell=True))
+    version = re.search("GCC:\s*\(.*\)\s*(\d+.\d+).\d+", s).group(1)
+    return version
+
+def get_tf_version():
+    import pkg_resources
+    s = pkg_resources.get_distribution("tensorflow-gpu").version
+    version = re.search("(\d+.\d+).\d+", s).group(1)
+    return version
+
+def get_gcc_compiler_version():
+    cmd = 'gcc --version | grep "gcc ("'
```

Comment:
I would give a user option to provide env variable pointing to his own, custom compiler. Now only the default one is tested.

Reply:
how would you name the variable?